cmake/machinecfg/Summit.cmake

```diff
@@ -18,20 +18,36 @@
 message(STATUS "Loading machine configuration for OLCF's Summit.\n"
   "Supported MACHINE_VARIANT includes 'cuda', 'mpi', and 'cuda-mpi'\n"
   "This configuration has been tested using the following modules: "
-  "module load cuda gcc cmake/3.14.2 python hdf5\n")
+  "module load cuda gcc cmake python hdf5\n")
 
 # common options
 set(Kokkos_ARCH_POWER9 ON CACHE BOOL "CPU architecture")
+set(PARTHENON_DISABLE_OPENMP ON CACHE BOOL "OpenMP support not yet tested in Parthenon.")
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Default release build")
+set(MACHINE_VARIANT "cuda-mpi" CACHE STRING "Default build for CUDA and MPI")
 
 # variants
 if (${MACHINE_VARIANT} MATCHES "cuda")
   set(Kokkos_ARCH_VOLTA70 ON CACHE BOOL "GPU architecture")
   set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Enable Cuda")
   set(CMAKE_CXX_COMPILER ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/bin/nvcc_wrapper CACHE STRING "Use nvcc_wrapper")
+else()
+  set(CMAKE_CXX_COMPILER g++ CACHE STRING "Use g++")
+  set(CMAKE_CXX_FLAGS "-fopenmp-simd -ffast-math -fprefetch-loop-arrays" CACHE STRING "Default opt flags")
 endif()
 
+# Setting launcher options independent of parallel or serial test as the launcher always
+# needs to be called from the batch node (so that the tests are actually run on the
+# compute nodes.
+set(TEST_MPIEXEC jsrun CACHE STRING "Command to launch MPI applications")
+set(TEST_NUMPROC_FLAG "-a" CACHE STRING "Flag to set number of processes")
+set(NUM_GPU_DEVICES_PER_NODE "6" CACHE STRING "6x V100 per node")
+set(PARTHENON_ENABLE_GPU_MPI_CHECKS OFF CACHE BOOL "Disable check by default")
+
 if (${MACHINE_VARIANT} MATCHES "mpi")
-  set(TEST_MPIEXEC jsrun CACHE STRING "Command to launch MPI applications")
-  set(TEST_NUMPROC_FLAG "-a" CACHE STRING "Flag to set number of processes")
+  # Use a single resource set on a node that includes all cores and GPUs.
+  # GPUs are automatically assigned round robin when run with more than one rank.
   list(APPEND TEST_MPIOPTS "-n" "1" "-g" "6" "-c" "42" "-r" "1" "-d" "packed" "-b" "packed:7" "--smpiargs='-gpu'")
```

Comment:
So if you are just running with mpi I'm guessing the gpu's just won't be called even though they are passed to mpi?

Reply:
What you do you refer to with "GPUs being called"?
These option ensure that there's a resource set for a node that includes all resources (all CPU cores and all GPUs).
The number of ranks that are launched in the tests is controlled via the `-a` flag and the number of GPUs controlled via the `kokkos-num-devices` argument.

Reply:
I'm saying why do you need "-g 6" if you are just running with the "mpi" variant, but I guess it doesn't really matter. If Kokkos is disabled anyway the gpus won't be utilized. Regardless, it's a non-issue.